src/server/cluster/cluster_config.cc

```diff
@@ -374,12 +381,47 @@ ClusterConfig::Node ClusterConfig::GetMasterNodeForSlot(SlotId id) const {
   return {};
 }
 
-ClusterConfig::ClusterShards ClusterConfig::GetConfig() const {
+ClusterShardConfigs ClusterConfig::GetConfig() const {
   return config_;
 }
 
 const SlotSet& ClusterConfig::GetOwnedSlots() const {
   return my_slots_;
 }
 
+static std::vector<MigrationInfo> GetMissingMigrations(const std::vector<MigrationInfo>& where,
+                                                       const std::vector<MigrationInfo>& which) {
+  std::vector<MigrationInfo> res;
+  for (const auto& w : where) {
+    if (find(which.begin(), which.end(), w) == which.end()) {
+      res.push_back(w);
+    }
+  }
+  return res;
+}
+
+std::vector<MigrationInfo> ClusterConfig::GetNewOutgoingMigrations(
+    std::shared_ptr<ClusterConfig> prev) const {
```

Comment:
Since you're not taking ownership over `prev`, it's better to accept it as `const&` to avoid the cost of atomic ops (inc/def ref)